contracts/marker/src/contract.rs

```diff
@@ -19,119 +19,119 @@ pub fn init(
-fn try_create_marker(coin: Coin) -> Result<HandleResponse<ProvenanceMsg>, ContractError> {
-    let msg = create_marker(coin.clone());
-    Ok(HandleResponse {
+fn try_create_marker(supply: Uint128, denom: String) -> HandleResponse<ProvenanceMsg> {
+    let msg = create_marker(supply.u128(), &denom);
+    HandleResponse {
         messages: vec![msg],
         attributes: vec![
             attr("action", "provwasm.contracts.marker.create"),
             attr("integration_test", "v2"),
-            attr("marker_denom", coin.denom),
-            attr("marker_supply", coin.amount.to_string()),
+            attr("marker_supply", supply),
+            attr("marker_denom", denom),
         ],
         data: None,
-    })
+    }
 }
 
 // Create and dispatch a message that will grant all permissions to a marker for an address.
-fn try_grant_marker_access(
-    denom: String,
-    address: HumanAddr,
-) -> Result<HandleResponse<ProvenanceMsg>, ContractError> {
-    let msg = grant_marker_access_all(denom.clone(), address.clone());
-    Ok(HandleResponse {
+fn try_grant_marker_access(denom: String, address: HumanAddr) -> HandleResponse<ProvenanceMsg> {
+    let msg = grant_marker_access_all(&denom, &address);
+    HandleResponse {
         messages: vec![msg],
         attributes: vec![
             attr("action", "provwasm.contracts.marker.grant_access"),
             attr("integration_test", "v2"),
             attr("marker_denom", denom),
-            attr("marker_addr", address.to_string()),
+            attr("marker_addr", address),
         ],
         data: None,
-    })
+    }
 }
 
 // Create and dispatch a message that will finalize a proposed marker.
-fn try_finalize_marker(denom: String) -> Result<HandleResponse<ProvenanceMsg>, ContractError> {
-    let msg = finalize_marker(denom.clone());
-    Ok(HandleResponse {
+fn try_finalize_marker(denom: String) -> HandleResponse<ProvenanceMsg> {
+    let msg = finalize_marker(&denom);
+    HandleResponse {
         messages: vec![msg],
         attributes: vec![
             attr("action", "provwasm.contracts.marker.finalize"),
             attr("integration_test", "v2"),
             attr("marker_denom", denom),
         ],
         data: None,
-    })
+    }
 }
 
 // Create and dispatch a message that will activate a finalized marker.
-fn try_activate_marker(denom: String) -> Result<HandleResponse<ProvenanceMsg>, ContractError> {
-    let msg = activate_marker(denom.clone());
-    Ok(HandleResponse {
+fn try_activate_marker(denom: String) -> HandleResponse<ProvenanceMsg> {
+    let msg = activate_marker(&denom);
+    HandleResponse {
         messages: vec![msg],
         attributes: vec![
             attr("action", "provwasm.contracts.marker.activate"),
             attr("integration_test", "v2"),
             attr("marker_denom", denom),
         ],
         data: None,
-    })
+    }
 }
 
 // Create and dispatch a message that will withdraw coins from a marker.
 fn try_withdraw_marker_coins(
```

Comment:
Curious about this api ... is there a way to withdraw coin that is not of the same denomination as the marker?

Reply:
because we need a moved reference to serialize as JSON for passing to the blockchain side.